control-plane/subcommand/server-acl-init/create_or_update_test.go

```diff
@@ -70,3 +70,95 @@ func TestCreateOrUpdateACLPolicy_ErrorsIfDescriptionDoesNotMatch(t *testing.T) {
 	require.NoError(err)
 	require.Equal(policyDescription, rereadPolicy.Description)
 }
+
+func TestCreateOrUpdateACLPolicy(t *testing.T) {
+	require := require.New(t)
+	ui := cli.NewMockUi()
+	k8s := fake.NewSimpleClientset()
+	cmd := Command{
+		UI:        ui,
+		clientset: k8s,
+		log:       hclog.NewNullLogger(),
+	}
+	cmd.init()
+	// Start Consul.
+	bootToken := "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee"
+	svr, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+		c.ACL.Enabled = true
+		c.ACL.Tokens.InitialManagement = bootToken
+	})
+	require.NoError(err)
+	defer svr.Stop()
+	svr.WaitForLeader(t)
+
+	// Get a Consul client.
+	consul, err := api.NewClient(&api.Config{
+		Address: svr.HTTPAddr,
+		Token:   bootToken,
+	})
+	require.NoError(err)
+	connectInjectRule, err := cmd.injectRules()
+	require.NoError(err)
+	aclReplRule, err := cmd.aclReplicationRules()
+	require.NoError(err)
+	policyDescription := "policy-description"
+	policyName := "policy-name"
+	policy, _, err := consul.ACL().PolicyCreate(&api.ACLPolicy{
+		Name:        "new-policy-name",
+		Description: "new-policy-desc",
+	}, nil)
+	require.NoError(err)
+	cases := []struct {
+		Name              string
+		ID                string
+		PolicyDescription string
+		PolicyName        string
+		Rules             string
+		Err               error
+		ExpPolicy         *api.ACLPolicy
+	}{
+		{
+			Name:              "create",
+			ID:                "",
+			PolicyDescription: policyDescription,
+			PolicyName:        policyName,
+			Rules:             connectInjectRule,
+			Err:               nil,
+			ExpPolicy: &api.ACLPolicy{
+				Name:        policyName,
+				Description: policyDescription,
+				Rules:       connectInjectRule,
+			},
+		},
+		{
+			Name:              "update",
+			ID:                policy.ID,
+			PolicyDescription: policy.Description,
+			PolicyName:        policy.Name,
+			Rules:             aclReplRule,
+			Err:               nil,
```

Comment:
`err` is `nil` for both cases so we can omit it as a test parameter.

Reply:
had kept this as there we can later add additional cases for the error flows( ex:- if updatePolicy/createPolicy returns err).
but we will have to mock acl. which will require code changes. 
 should i remove it ?

Reply:
Yes please remove. We can add it when we're adding those test cases in the future.